### DISCUSSION TITLE
The choice between removing highly contaminated cells and using the corrected count matrix

### DISCUSSION THREAD

**Discussion by LiuCanidk** [campbio/decontX#25] [General]
Hi, thanks for developing this convenient tools!

I have some problem on QC of my single cell expression data after performing the decontX analysis. As we know, the decontX object provides results of both a corrected count matrix and the column of contamination score. So I guess and from what I found in searing engine, there are two potential ways to perform quality control on my own dataset: i) just use the corrected count matrix and create another new seurat object, as the tutorial showed. ii) set a threshold for contamination score, and remove highly contaminated cells.

I'd like to ask which way should I go? or should I perform both of them to assure the purity of my dataset? i.e., remove the highly contaminated cells and also use the corrected matrix for the remaining cells.

Any reply or discussion would be greatly appreciated!

**Comment by joshua-d-campbell** [comment]
Hi @LiuCanidk, thanks for the good question. We generally use the corrected counts if it helps remove contaminating genes from down-stream differential expression analyses or if it helps improve the clarity of the UMAP or the specificity of the dotplots. We usually only filter out cells cells with high contamination if they are adversely affecting the downstream analysis. For example some clusters of cells may be drive by highly contaminated cells or the highly contaminated cells negatively affect downstream plotting (e.g. UMAPs). This can especially occur with single nucleus sequencing data. Sometimes we apply a threshold (e.g. remove cells with >0.30 contamination levels) or just identify the clusters with enrichment for contaminated cells and remove the whole cluster. 

Hopefully this helps as a guide when to use either (or both) approaches for improving your data. I'm going to convert this to a Discussion so others can benefit.

**Reply by LiuCanidk** [reply]
> Hi @LiuCanidk, thanks for the good question. We generally use the corrected counts if it helps remove contaminating genes from down-stream differential expression analyses or if it helps improve the clarity of the UMAP or the specificity of the dotplots. We usually only filter out cells cells with high contamination if they are adversely affecting the downstream analysis. For example some clusters of cells may be drive by highly contaminated cells or the highly contaminated cells negatively affect downstream plotting (e.g. UMAPs). This can especially occur with single nucleus sequencing data. Sometimes we apply a threshold (e.g. remove cells with >0.30 contamination levels) or just identify the clusters with enrichment for contaminated cells and remove the whole cluster.
> 
> Hopefully this helps as a guide when to use either (or both) approaches for improving your data. I'm going to convert this to a Discussion so others can benefit.

Thanks for the reply!
